src/v/coproc/tests/fixtures/fiber_mock_fixture.cc

```diff
@@ -0,0 +1,199 @@
+/*
+ * Copyright 2021 Vectorized, Inc.
+ *
+ * Use of this software is governed by the Business Source License
+ * included in the file licenses/BSL.md
+ *
+ * As of the Change Date specified in that file, in accordance with
+ * the Business Source License, use of this software will be governed
+ * by the Apache License, Version 2.0
+ */
+
+#include "coproc/tests/fixtures/fiber_mock_fixture.h"
+
+#include "coproc/api.h"
+#include "coproc/tests/utils/batch_utils.h"
+
+#include <seastar/core/coroutine.hh>
+
+namespace {
+
+class counting_consumer {
+public:
+    ss::future<ss::stop_iteration> operator()(const model::record_batch& rb) {
```

Comment:
A previous commit has:
```cpp
        auto data = co_await model::consume_reader_to_memory(
          std::move(*e.reader), model::no_timeout);
        if (!data.empty()) {
```
This looks like it might be useful and cheaper.

Reply:
Oh, I meant `counting_consumer` looks nicer. No need to change it.